seahorn/lib/CMakeLists.txt

```diff
@@ -17,14 +17,21 @@ add_library(sea_proofs
   sea_allocators.c
   allocator_override.c
   bcmp.c
-  sea_string.c
   error_override.c)
 
 target_compile_definitions(sea_proofs PUBLIC
   MAX_BUFFER_SIZE=${MAX_BUFFER_SIZE}
   MAX_STRING_LEN=${MAX_STRING_LEN})
 sea_attach_bc(sea_proofs)
 
+# string proof helpers
+add_library(str_proofs
+  sea_string.cc)
+
+target_compile_definitions(str_proofs PUBLIC
+    MAX_BUFFER_SIZE=${MAX_BUFFER_SIZE}
```

Comment:
probably need to make the bounds available through API somehow else.
At least add a comment here for now.

We want to avoid relying on macros. Need to think what the best solution is for templates.